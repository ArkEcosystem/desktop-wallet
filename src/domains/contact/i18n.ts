export const translations: { [key: string]: any } = {
	CONTACTS_PAGE: {
		ADD_CONTACT: "Add Contact",
		ADD_CONTACT_MESSAGE: "Add your most frequent contacts for fast, easy payments",
		TITLE: "My Contacts",
		SUBTITLE: "Manage your frequent contacts",
	},
	CONTACT_FORM: {
		ADD_ADDRESS: "Add address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		DELETE_CONTACT: "Delete Contact",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
	MODAL_SEARCH_CONTACT: {
		TITLE: "Search Contact",
		DESCRIPTION: "Find the contact you need quickly and conveniently",
		PLACEHOLDER: "Search...",
	},
	VALIDATION: {
		MAXLENGTH_ERROR: "The Contact Name should have less than {{maxLength}} characters.",
		CONTACT_NAME_EXISTS: "Contact with name {{name}} already exists.",
		CONTACT_NAME_EXISTS_SUFFIX: "already exists",
	},
};
