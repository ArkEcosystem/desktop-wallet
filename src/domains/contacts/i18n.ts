export const translations: { [key: string]: any } = {
	CONTACT_FORM: {
		ADD_ADDRESS: "Add Address",
		ADDRESS: "Address",
		ADDRESSES: "Addresses",
		NAME: "Name",
		NETWORK: "Network",
	},
	MODAL_CREATE_CONTACT: {
		TITLE: "Add Contact",
		DESCRIPTION: "Enter your contacts name and addresses and add them to your list.",
	},
	MODAL_DELETE_CONTACT: {
		TITLE: "Delete Contact",
		DESCRIPTION: "Do you really want to delete this contact? Once deleted, you will not be able to restore it.",
	},
	MODAL_UPDATE_CONTACT: {
		TITLE: "Edit Contact",
	},
};
