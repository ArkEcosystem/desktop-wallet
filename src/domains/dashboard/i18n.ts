export const translations: { [key: string]: any } = {
	FILTER_WALLETS: {
		NETWORK: {
			TITLE: "Filter Network",
			DESCRIPTION: "Select the types of networks",
		},
		WALLETS: {
			TITLE: "Your Wallets",
			DESCRIPTION: "Select the display of your wallets",
		},
		CHART: {
			TITLE: "Portfolio Chart",
			DESCRIPTION: "You can hide this unit if you like",
		},
		TRANSACTION_HISTORY: {
			TITLE: "Transaction History",
			DESCRIPTION: "You can hide this unit if you like",
		},
	},

	TRANSACTION_HISTORY: {
		TITLE: "Transactions History",
		EMPTY_TEXT:
			"This will display the history of your transactions. But you don't have more than one transaction at the moment.",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		IMPORT: "Import",
	},

	DASHBOARD_PAGE: {
		CHART: {
			TITLE: "Portfolio Chart",
			PERCENTAGES_LABEL: "Total portfolio",
		},
	},
};
