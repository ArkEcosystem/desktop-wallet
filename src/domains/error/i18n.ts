export const translations: { [key: string]: any } = {
	APPLICATION: {
		TITLE: "Something went wrong",
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		RELOAD: "Reload",
	},
	OFFLINE: {
		TITLE: "Cannot Connect",
		DESCRIPTION: "Check your internet connection if it doesn't work, check it again.",
	},
};
