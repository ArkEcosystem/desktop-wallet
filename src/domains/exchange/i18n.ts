export const translations: { [key: string]: any } = {
	MODAL_ADD_EXCHANGE: {
		TITLE: "Exchange",
		DESCRIPTION:
			"Buy ARK directly from the ARK Desktop Wallet or through one of 20+ exchanges supplying ARK tokens using BTC, ETH or other trading pairs.",
	},
};
