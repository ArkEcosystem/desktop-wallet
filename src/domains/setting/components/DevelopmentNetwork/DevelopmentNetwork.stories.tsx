import { boolean } from "@storybook/addon-knobs";
import React from "react";

import { DevelopmentNetwork } from "./DevelopmentNetwork";

export default { title: "Domains / Setting / Components / DevelopmentNetwork" };

export const Default = () => (
	<DevelopmentNetwork
		isOpen={boolean("Is Open", true)}
		onClose={() => alert("closed")}
		onCancel={() => alert("canceled")}
		onContinue={() => alert("continue")}
	/>
);
