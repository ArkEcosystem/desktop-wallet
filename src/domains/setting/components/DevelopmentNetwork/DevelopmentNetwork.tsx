import { Button } from "app/components/Button";
import { Modal } from "app/components/Modal";
import React from "react";
import { useTranslation } from "react-i18next";

type DevelopmentNetworkProps = {
	isOpen: boolean;
	onClose?: () => void;
	onCancel?: () => void;
	onContinue?: () => void;
};

export const DevelopmentNetwork = ({ isOpen, onClose, onCancel, onContinue }: DevelopmentNetworkProps) => {
	const { t } = useTranslation();

	return (
		<Modal
			title={t("SETTINGS.MODAL_DEVELOPMENT_NETWORK.TITLE")}
			description={t("SETTINGS.MODAL_DEVELOPMENT_NETWORK.DESCRIPTION")}
			size="xl"
			isOpen={isOpen}
			onClose={onClose}
		>
			<div className="flex justify-end mt-8 space-x-3">
				<Button variant="secondary" onClick={onCancel} data-testid="DevelopmentNetwork__cancel-button">
					{t("COMMON.CANCEL")}
				</Button>

				<Button onClick={onContinue} data-testid="DevelopmentNetwork__continue-button">
					{t("COMMON.CONTINUE")}
				</Button>
			</div>
		</Modal>
	);
};

DevelopmentNetwork.defaultProps = {
	isOpen: false,
};
