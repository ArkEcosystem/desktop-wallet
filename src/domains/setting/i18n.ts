export const translations: { [key: string]: any } = {
	TITLE: "Settings",

	GENERAL: {
		MENU_ITEM: "General",
		TITLE: "Profile Settings",
		SUBTITLE: "Customize your wallet to suit your needs.",
		PERSONAL: {
			TITLE: "Personal Details",
			PROFILE_IMAGE: "Profile Image",
			DELETE_AVATAR: "Delete Avatar",
			UPLOAD_AVATAR: "Upload Avatar",
			NAME: "Name",
			PASSWORD: "Password",
			CONFIRM_PASSWORD: "Confirm Password",
			PASSPHRASE_LANGUAGE: "Passphrase Language",
			CURRENCY: "Currency",
			LANGUAGE: "Language",
			MARKET_PROVIDER: "Market Provider",
			TIME_FORMAT: "Time Format",
			VALIDATION: {
				NAME_EXISTS: "Profile name already exists",
			},
		},
		SECURITY: {
			TITLE: "Security",
			SCREENSHOT_PROTECTION: {
				TITLE: "Screenshot Protection",
				DESCRIPTION: "This protection will protect your money from unwanted Screenshot you PC.",
			},
			ADVANCED_MODE: {
				TITLE: "Advanced Mode",
				DESCRIPTION:
					"Allows for installing files from direct URLs. You assume any associated risk by enabling this feature.",
			},
			AUTOMATIC_SIGN_OUT_PERIOD: {
				TITLE: "Automatic Sign Out Period",
			},
		},
		OTHER: {
			TITLE: "Other",
			DARK_THEME: {
				TITLE: "Dark Theme",
				DESCRIPTION: "When enabled, changes to a dark, high contrast theme.",
			},
			DEVELOPMENT_NETWORKS: {
				TITLE: "Development Networks",
				DESCRIPTION: "When enabled, allows application to add development and test networks.",
			},
			ERROR_REPORTING: {
				TITLE: "Error Reporting",
				DESCRIPTION: "Automatically send reports to help to improve our products.",
			},
			TRANSACTION_HISTORY: {
				TITLE: "Portfolio Transaction History",
				DESCRIPTION: "Enables transaction history on Portfolio page",
			},
		},
		SUCCESS: "Your profile settings have been updated",
		ERROR: "Your profile settings could not be updated",
	},

	PASSWORD: {
		MENU_ITEM: "Password",
		TITLE: "Profile Password",
		SUBTITLE_CREATE: "Provide a password below to secure your user profile.",
		SUBTITLE_UPDATE: "Change the password used to access your profile below.",
		BUTTON: {
			CREATE: "Set Password",
			UPDATE: "Change Password",
		},
		CURRENT: "Current Password",
		PASSWORD_1: "New Password",
		PASSWORD_2: "Confirm New Password",
		ERROR: {
			FALLBACK: "Something went wrong.",
			MISMATCH: "The current password does not match.",
		},
		SUCCESS: "Your password has been successfully changed.",
	},

	PEERS: {
		MENU_ITEM: "Peers",
		TITLE: "Peer Settings",
		SUBTITLE: "Customize your network peer settings.",
		BROADCAST_TRANSACTIONS: {
			TITLE: "Broadcast to Multiple Peers",
			DESCRIPTION: "Toggle to send transactions to multiple network peers. (Recommended)",
		},
		CUSTOM_PEERS: {
			TITLE: "Use Custom Peers",
			DESCRIPTION:
				"Customize your individual peers by network. <1/> Note: Only use trusted peers. Using an unknown peer may put your funds at risk.",
		},
		CRYPTOASSET: "Cryptoasset",
		ADD_PEER: "Add Peer",
		EDIT_PEER: "Edit Peer",
		NAME: "Name",
		PEER_IP: "Peer IP",
		TYPE: "Type",
		SUCCESS: "Your peer settings have been updated",
		VALIDATION: {
			HOST_EXISTS: "The host already exists for this network",
			NOT_VALID: "The '{{field}}' is not valid",
			NO_PROTOCOL: "The '{{field}}' does not have 'http://' or 'https://'",
		},
	},

	PLUGINS: {
		MENU_ITEM: "Plugins",
		TITLE: "Plugin Settings",
		SUBTITLE: "Customize your wallet to suit your needs.",
		PLUGIN_SOURCE: {
			TITLE: "Plugin Source",
			DESCRIPTION: "Turn this feature on, you can upload plugins to your wallet from third-party sources.",
			LOAD_FROM: "Load plugins from",
		},
		ADD_PLUGIN: "Add Plugin",
		SUCCESS: "Your plugin settings have been updated",
	},

	EXPORT: {
		MENU_ITEM: "Export",
		TITLE: "Export Profile",
		SUBTITLE:
			"Your exported Profile will not contain your mnemonic passphrases, only addresses and respective names.",
		EXCLUDE_WALLETS_WITHOUT_NAME: "Exclude wallets without a name",
		EXCLUDE_EMPTY_WALLETS: "Exclude empty wallets",
		EXCLUDE_LEDGER_WALLETS: "Exclude Ledger wallets",
		ADD_INFORMATION_ABOUT_THE_NETWORK: "Add information about the network",
		SAVE_GENERAL_CUSTOMIZATIONS: "Save general settings customizations",
		SUCCESS: "Your settings have been exported successfully",
	},

	MODAL_ADVANCED_MODE: {
		TITLE: "Advanced Mode Disclaimer",
		DISCLAIMER:
			'By enabling Advanced Mode you hereby assume the risk associated with downloading files and installing said files from a direct URL link. The aforementioned links have neither been tested nor approved by ARK.io and ARK SCIC.\n\nWe make no warranties, expressed or implied, as to the sustainability, availability, security, of these URLs. We are not responsible for any consequences which may occur from downloading, viewing, or installing files directly from a URL or feature that is activated via toggling on Advanced Mode. ARK.io and ARK SCIC shall bear no liability for any loss suffered by users who bypass certain measures by toggling on Advanced Mode.\n\nBy clicking "I Accept" you acknowledge that enabling Advanced Mode may result in you downloading, viewing, or installing content that may have bugs, glitches, lack of functionality, or can cause damage to your machine or result in the loss of data.',
	},

	MODAL_DEVELOPMENT_NETWORK: {
		TITLE: "Development Network",
		DESCRIPTION:
			"Disabling this setting will hide your wallets associated with development networks. Are you sure you want to continue?",
	},

	MODAL_CUSTOM_PEER: {
		TITLE: "Custom Peer",
	},

	MODAL_DELETE_PEER: {
		TITLE: "Delete Peer",
		DESCRIPTION: "Do you really want to delete this peer? Once deleted, you will not be able to restore it.",
	},
};
