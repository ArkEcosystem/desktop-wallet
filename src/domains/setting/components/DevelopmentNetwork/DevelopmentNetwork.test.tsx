import React from "react";
import { render } from "testing-library";

import { translations } from "../../i18n";
import { DevelopmentNetwork } from "./DevelopmentNetwork";

describe("DevelopmentNetwork", () => {
	it("should not render if not open", () => {
		const { asFragment, getByTestId } = render(<DevelopmentNetwork isOpen={false} />);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render a modal", () => {
		const { asFragment, getByTestId } = render(<DevelopmentNetwork isOpen={true} />);

		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_DEVELOPMENT_NETWORK.TITLE);
		expect(getByTestId("modal__inner")).toHaveTextContent(translations.MODAL_DEVELOPMENT_NETWORK.DESCRIPTION);
		expect(asFragment()).toMatchSnapshot();
	});
});
