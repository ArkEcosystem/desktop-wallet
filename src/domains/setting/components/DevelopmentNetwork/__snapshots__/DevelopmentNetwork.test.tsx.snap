// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevelopmentNetwork should not render if not open 1`] = `<DocumentFragment />`;

exports[`DevelopmentNetwork should render a modal 1`] = `
<DocumentFragment>
  <div
    class="flex fixed inset-0 z-50 justify-center items-center w-full h-full overflow-overlay"
  >
    <div
      class="fixed z-50 w-full h-full bg-black opacity-50"
      data-testid="modal__overlay"
    />
    <div
      class="Modal__ModalContainer-sc-1pv7yg7-0 gDeenD absolute left-0 right-0 z-50 flex flex-col p-10 mx-auto overflow-hidden rounded-2xl bg-theme-background shadow-2xl "
      data-testid="modal__inner"
    >
      <div
        class="absolute top-0 right-0 z-50 mt-5 mr-5 rounded transition-all duration-100 ease-linear bg-theme-primary-100 hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
      >
        <button
          class="Button__StyledButton-mdtvtf-0 fHYGyx w-11 h-11"
          data-testid="modal__close-btn"
          type="button"
        >
          <div
            class="flex relative items-center space-x-2"
          >
            <div
              class="sc-bdfBwQ bfBhNR"
              height="14"
              width="14"
            >
              <svg>
                cross-slim.svg
              </svg>
            </div>
          </div>
        </button>
      </div>
      <div
        class="relative"
      >
        <h2
          class="mb-0 text-3xl font-bold"
        >
          Development Network
        </h2>
        <div
          class="flex-1"
        >
          <div
            class="mt-1 text-theme-secondary-text whitespace-pre-line"
          >
            Disabling this setting will hide your wallets associated with development networks. Are you sure you want to continue?
          </div>
          <div
            class="flex justify-end mt-8 space-x-3"
          >
            <button
              class="Button__StyledButton-mdtvtf-0 iUSBMV"
              data-testid="DevelopmentNetwork__cancel-button"
              type="button"
            >
              <div
                class="flex relative items-center space-x-2"
              >
                Cancel
              </div>
            </button>
            <button
              class="Button__StyledButton-mdtvtf-0 ePkfgQ"
              data-testid="DevelopmentNetwork__continue-button"
              type="button"
            >
              <div
                class="flex relative items-center space-x-2"
              >
                Continue
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
